Plugin/Scripting/CodeLiteLUA.cpp

```diff
@@ -0,0 +1,184 @@
+{
+enum MessageType {
+    kInfo,
+    kWarn,
+    kError,
+};
+}
+
+CodeLiteLUA& CodeLiteLUA::Get()
+{
+    static CodeLiteLUA codelite_lua;
+    return codelite_lua;
+}
+
+void CodeLiteLUA::Reset() { m_menu_items.clear(); }
+
+void CodeLiteLUA::Initialise()
+{
+    auto& self = Get();
+    self.m_state = luaL_newstate();
+    luaL_openlibs(self.m_state);
+
+    try {
+        luabridge::getGlobalNamespace(self.m_state)
+            .beginNamespace("codelite")
+            .addFunction("message_box", &CodeLiteLUA::message_box)
+            .addFunction("add_menu_item", &CodeLiteLUA::add_menu_item)
+            .endNamespace();
+
+    } catch (const std::exception& e) {
+        self.m_state = nullptr; // it's ok to have the memory leak
+        clERROR() << "Error occurred while initialising LUA." << e.what() << endl;
+        return;
+    }
+    self.InitialiseInternal();
+    EventNotifier::Get()->Bind(wxEVT_FILE_SAVED, &CodeLiteLUA::OnFileSaved, &self);
+}
+
+void CodeLiteLUA::InitialiseInternal()
+{
+    Reset();
+
+    // Load and compile CodeLite's main lua script.
+    auto options = WriteOptions{.force_global = true};
+    wxString codelite_lua = FileManager::GetSettingFileFullPath("codelite.lua", options);
+    clDEBUG() << "Reading file:" << codelite_lua << endl;
+    auto content = FileManager::ReadSettingsFileContent(codelite_lua, options);
+    if (!content.has_value()) {
+        clDEBUG() << "File:" << codelite_lua << "does not exist" << endl;
+        return;
+    }
+
+    clDEBUG() << "File:" << codelite_lua << "content:\n" << content.value() << endl;
+    auto status = Run(content.value());
+    if (!status.ok()) {
+        wxString errmsg;
+        errmsg << _("Failed to run LUA file: ") << codelite_lua << "\n" << status.message();
+        ::wxMessageBox(errmsg, "CodeLite", wxICON_ERROR | wxOK | wxCENTER);
+        return;
+    }
+    clDEBUG() << "Successfully executed script file:" << codelite_lua << endl;
+}
+
+CodeLiteLUA::CodeLiteLUA() {}
+
+CodeLiteLUA::~CodeLiteLUA()
+{
+    // free the state
+    m_menu_items.clear();
+    if (m_state) {
+        lua_close(m_state);
+        m_state = nullptr;
+    }
+}
+
+void CodeLiteLUA::message_box(const std::string& message, int type)
+{
+    switch (static_cast<MessageType>(type)) {
+    case MessageType::kWarn:
+        ::wxMessageBox(wxString::FromUTF8(message), "CodeLite Lua", wxOK | wxCENTER | wxICON_WARNING);
+        break;
+    case MessageType::kError:
+        ::wxMessageBox(wxString::FromUTF8(message), "CodeLite Lua", wxOK | wxCENTER | wxICON_ERROR);
+        break;
+    default:
+    case MessageType::kInfo:
+        ::wxMessageBox(wxString::FromUTF8(message), "CodeLite Lua", wxOK | wxCENTER | wxICON_INFORMATION);
+        break;
+    }
+}
+
+void CodeLiteLUA::add_menu_item(const std::string& menu_name, const std::string& label, luabridge::LuaRef action)
+{
+    auto& self = Get();
+
+    clDEBUG() << "Adding item:" << label << "for menu:" << menu_name << endl;
```

Comment:
```suggestion
    clDEBUG() << "Adding item: " << label << " for menu:" << menu_name << endl;
```

Reply:
The logging system of CodeLite adds the space automatically, so it is not needed. We should enhance it to support `std::setw(...)` and other methods from the STL.